### PR TITLE
Add licenseSets

### PR DESCRIPTION
**Pull request by jkrech**
Add licenseSets to specification as requested in #24

### CODE REVIEW

File: doxygen/pack.dxy [hunk 1]
```diff
@@ -867,6 +867,7 @@ INPUT           =   src/General.txt \
                     src/config_wizard.txt \
                     src/flash_algorithms.txt \
                     src/conditions_schema.txt \
+					src/licenses_schema.txt \
```
Comment (LMESTM) [inline doxygen/pack.dxy:870]: alignment issue ?

File: doxygen/src/General.txt [hunk 2]
```diff
@@ -83,6 +83,13 @@ Pack\\Tutorials | Tutorials for \ref cp_Packs "Creating Packs" |
     <th>Version</th>
     <th>Description</th>
   </tr>
+  <tr>
+    <td>1.7.8</td>
+    <td>
+     - added 'includeAsm', 'includeC', 'includeCpp', 'includeLd' to FileCategoryType
```
Comment (LMESTM) [inline doxygen/src/General.txt:89]: is this related to licenseSets ?

Comment (LMESTM) [inline doxygen/src/General.txt:89]: is this related to licenseSets ?

File: doxygen/src/apis_schema.txt [hunk 3]
```diff
@@ -171,11 +172,17 @@ by either implementing or using a set of functions, data types, and definitions.
   </tr>
   <tr>
     <td>Capiversion</td>
-    <td>Version of this component. Is a mandatory part of the component ID.  The version format is described in \ref VersionType "Version Type".
+    <td>Version of the api. Is a mandatory part of the api ID.  The version format is described in \ref VersionType "Version Type".
```
Comment (LMESTM) [inline doxygen/src/apis_schema.txt:175]: same here - does seems to be related to licenceSets, should probably be fixed in separate PR / commit.

Comment (LMESTM) [inline doxygen/src/apis_schema.txt:175]: same here - does not seem to be related to licenceSets, should probably be fixed in separate PR / commit.
Also applies to other lines in this commit

**Review by LMESTM — COMMENTED** [review]
Overall looks good, except some changes seem maybe not to be related

**Comment by JonatanAntoni** [comment]
How do we take this one on?

File: doxygen/src/General.txt (hunk 2 above)
Comment (JonatanAntoni) [reply doxygen/src/General.txt]: nope, removed

File: doxygen/src/apis_schema.txt (hunk 3 above)
Comment (JonatanAntoni) [reply doxygen/src/apis_schema.txt:175]: removed

File: doxygen/src/apis_schema.txt [hunk 4]
```diff
@@ -171,11 +172,17 @@ by either implementing or using a set of functions, data types, and definitions.
...
 	</td>
     <td>VersionType</td>
     <td>optional</td>
   </tr>
+  <tr>
+    <td>licenseSet</td>
+	<td>Reference to the licenseSet with the given identifier \token{ID} listed in the licenseSets section of the package description.
+	    A licenseSet describes one or more license files which in this case govern the api.</td>
```
[the remark this answers is not in the document]
Comment (JonatanAntoni) [reply doxygen/src/apis_schema.txt:183]: This may need your attention, @jkrech.

[the remark this answers is not in the document]
Comment (jkrech) [reply doxygen/src/apis_schema.txt:183]: @LMESTM just to be clear, this is the <api> definition section and since such an API maybe governed by a license which is different from the implementations of such an API, I think we should allow to specify a license reference for the api.

e.g. CMSIS-Driver USART API is specified and the header file is stored in  ARM.CMSIS pack using Apache 2.0 license
The implementation of a CMSIS-Driver USART for a device maybe governed by a different license in respective DFP.

[the remark this answers is not in the document]
Comment (LMESTM) [reply doxygen/src/apis_schema.txt:183]: @jkrech was my mistake when reading, I though this was the licence for any component ... as this is the api section yes the description is valid. Please discard my comment
